scss/util/_breakpoint.scss

```diff
@@ -166,7 +168,7 @@ $breakpoint-classes: (small medium large) !default;
     // Otherwise, wrap the content in a media query
     @else {
       // For named breakpoints less than or equal to $print-breakpoint, add print to the media types
-      @if $bp != null and $bp <= $pbp {
+      @if ($bp != null and $bp == $pbp and ($dir == '' or $dir == 'up' or $dir == 'only')) or ($bp != null and $bp < $pbp and ($dir == '' or $dir == 'up'))  or ($bp != null and $bp >= $pbp and ($dir == 'down')) {
```

Comment:
I think we should move this to a cleaner "breakpoint compairaison" function. Putting all the cases here in a `if` is really confusing.

Reply:
Definitely. Could need some assistance here as I've not often written mixins and functions.

Reply:
I'll take care of this.

Reply:
Thanks =)

Reply:
... this weekend :)

Reply:
No problem.

Reply:
@DanielRuf @ncoden I'm not sure if this line has to be changed at all... what is the goal?

The comment says 'for named breakpoints less than or equal to $print-breakpoint' what I interpreted as starting point . The original `@include breakpoint(large)` was actually a `@include breakpoint(large up)` and thus also affects breakpoints greater than large.

So is the goal that `breakpoint(large only)` and `breakpoint(large down)` do affect print but `breakpoint(large)` does not? (although this would be a breaking change)
If not the condition `$bp != null and $bp <= $pbp` doesn't need to be changed imo

Reply:
I had some trouble to retrieve the problem here and the purpose of this change, but I think I get it. 

If I understand `$print-breakpoint` correctly, it must be used to prevent `print` to consider highest breakpoints because despite the high resolution of the paper (which is good for reading), this is not a good support for small _interfaces_ and interactive elements. Paper should almost be considered as a _tablet_ giving a fake resolution (like iPads, but _we_ are faking the resolution).

According to this, the objective is to make print ignoring all styles applied to _highest breakpoints only and not below_. This means that, for `$print-breakpoint: large;` on print:
* `large up/only/down`, `medium up/only/down`, `xlarge down` should be printable because their styles are applied to a _large_ (or smaller) screen so we should consider them as displayable on a _large_ (or smaller) print support.
* `xlarge only/up` should be not printable like before.

Reply:
Ok @ncoden so the rule would be to generate print if the breakpoint affects the print-breakpoint (or smaller).
This means the current condition only needs to be extended so 'down' always generates print.

Since @DanielRuf has already added `$dir` I would simply change the condition to
```sass
@if $bp != null and ($bp <= $pbp or $dir == down) {
```

Reply:
Yes, and we can add an explantation comment and a link to this. @DanielRuf What do you think ?

Reply:
Sounds good. I might have thought to complex. Not sure what I covered that would not be covered by the simple `or ` condition. 🤔